src/coord/src/catalog.rs

```diff
@@ -774,6 +774,17 @@ impl Catalog {
         self.by_id.insert(entry.id, entry);
     }
 
+    pub fn set_source_connector(&mut self, id: GlobalId, source_connector: &SourceConnector) {
+        let item = &mut self.by_id.get_mut(&id).unwrap().item;
+        *item = match item.clone() {
+            CatalogItem::Source(mut source) => {
+                source.connector = source_connector.clone();
+                CatalogItem::Source(source)
+            }
+            _ => unreachable!(),
+        }
+    }
```

Comment:
Shouldn't need to clone item if you do this!

```rust
    pub fn set_source_connector(&mut self, id: GlobalId, source_connector: SourceConnector) {
        match &mut self.by_id.get_mut(&id).unwrap().item {
            CatalogItem::Source(source) => {
                source.connector = source_connector;
            }
            _ => unreachable!(),
        }
    }
```

I also changed the func signature to take `SourceConnector` by value. As a general Rust API design principle, if you need an owned copy of something, take it by value rather than taking a ref that you clone internally. This can save clones if the caller has a spare copy handy—and if the caller doesn't have a copy handy, cloning in the caller instead of the callee doesn't make you any worse off.

Here's the whole diff if you like. (I've learned to prepare things before I suggest them because 50% of the time Rust just totally rejects my suggestion for reasons I didn't forsee.)

```diff
diff --git a/src/coord/src/catalog.rs b/src/coord/src/catalog.rs
index cdfb916ec..ff2e894e5 100644
--- a/src/coord/src/catalog.rs
+++ b/src/coord/src/catalog.rs
@@ -774,12 +774,10 @@ impl Catalog {
         self.by_id.insert(entry.id, entry);
     }
 
-    pub fn set_source_connector(&mut self, id: GlobalId, source_connector: &SourceConne>
-        let item = &mut self.by_id.get_mut(&id).unwrap().item;
-        *item = match item.clone() {
-            CatalogItem::Source(mut source) => {
-                source.connector = source_connector.clone();
-                CatalogItem::Source(source)
+    pub fn set_source_connector(&mut self, id: GlobalId, source_connector: SourceConnec>
+        match &mut self.by_id.get_mut(&id).unwrap().item {
+            CatalogItem::Source(source) => {
+                source.connector = source_connector;
             }
             _ => unreachable!(),
         }
diff --git a/src/coord/src/coord.rs b/src/coord/src/coord.rs
index 87a83ba74..804899e53 100644
--- a/src/coord/src/coord.rs
+++ b/src/coord/src/coord.rs
@@ -275,7 +275,7 @@ where
                         // If we got back a new connector, we need to take some
                         // additional action.
                         if let Some(connector) = connector {
-                            coord.enable_source_connector_persistence(*id, &connector);
+                            coord.enable_source_connector_persistence(*id, connector);
                         }
                     }
                 }
@@ -1222,7 +1222,7 @@ where
                     self.ship_dataflow(self.build_index_dataflow(index_id));
                 }
 
-                self.enable_source_connector_persistence(source_id, &source.connector);
+                self.enable_source_connector_persistence(source_id, source.connector);
                 Ok(ExecuteResponse::CreatedSource { existed: false })
             }
             Err(_) if if_not_exists => Ok(ExecuteResponse::CreatedSource { existed: tru>
@@ -2288,7 +2288,7 @@ where
     fn enable_source_connector_persistence(
         &mut self,
         id: GlobalId,
-        source_connector: &SourceConnector,
+        source_connector: SourceConnector,
     ) {
         if let SourceConnector::External { connector, .. } = &source_connector {
             if connector.persistence_enabled() {
```